src/devices/bus/coco/meb_rtime.cpp

```diff
@@ -0,0 +1,173 @@
+	//  disto_rtime_device - constructor
+	//-------------------------------------------------
+
+	disto_rtime_device::disto_rtime_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
+		: device_t(mconfig, DISTOMEB_RTIME, tag, owner, clock)
+		, device_distomeb_interface(mconfig, *this)
+		, m_rtc(*this, RTC_TAG)
+		, m_rtc_address(0)
+		, m_double_write(0)
+		, m_centronics(*this, CENTRONICS_TAG)
+		, m_latch(*this, LATCH_TAG)
+		, m_centronics_busy(0)
+	{
+	}
+
+	//-------------------------------------------------
+	//  device_start - device-specific startup
+	//-------------------------------------------------
+
+	void disto_rtime_device::device_start()
+	{
+		// save state
+		save_item(NAME(m_rtc_address));
+		save_item(NAME(m_double_write));
+		save_item(NAME(m_centronics_busy));
+	}
+
+	//-------------------------------------------------
+	//  device_add_mconfig - add device configuration
+	//-------------------------------------------------
+
+	void disto_rtime_device::device_add_mconfig(machine_config &config)
+	{
+		MSM6242(config, m_rtc, XTAL(32'768));
+
+		CENTRONICS(config, m_centronics, centronics_devices, "printer");
+		m_centronics->busy_handler().set(FUNC(disto_rtime_device::busy_w));
+
+		OUTPUT_LATCH(config, m_latch);
+		m_centronics->set_output_latch(*m_latch);
+	}
+
+	//-------------------------------------------------
+	//  meb_read
+	//-------------------------------------------------
+
+	u8 disto_rtime_device::meb_read(offs_t offset)
+	{
+		u8 result= 0;
+
+		switch(offset)
+		{
+			case 0x00:  /* FF50 */
+				result = m_rtc->read(m_rtc_address & 0x0f);
+				break;
+
+			case 0x02:	/* FF52 */
+				result = m_centronics_busy << 7;
+		}
+
+		return result;
+	}
+
+
+	//-------------------------------------------------
+	//    meb_write
+	//-------------------------------------------------
+
+	void disto_rtime_device::meb_write(offs_t offset, u8 data)
+	{
+		switch(offset)
+		{
+			case 0x00: /* FF50 */
+				m_rtc->write(m_rtc_address & 0x0f, data);
+				break;
+
+			case 0x01: /* FF51 */
+				m_rtc_address = data;
+				break;
+
+			case 0x02: /* FF52 */
+				if ((m_rtc_address == data) && !m_double_write)
+				{
+					m_double_write = 1;
+					m_latch->write(data);
+					m_centronics->write_strobe(1);
+					m_centronics->write_strobe(0);
+				}
+
+				m_rtc_address = data;
+				break;
+		}
+
+		m_double_write = 0;
+	}
+
+
+	//-------------------------------------------------
+	//    busy_w - centronics busy call back
+	//-------------------------------------------------
+
+	WRITE_LINE_MEMBER( disto_rtime_device::busy_w )
+	{
+		m_centronics_busy = state;
+	}
+}
+
+	//**************************************************************************
+	//  GLOBAL VARIABLES
+	//**************************************************************************
+
+	DEFINE_DEVICE_TYPE_PRIVATE(DISTOMEB_RTIME, device_distomeb_interface, disto_rtime_device, "distomeb_rtime", "Disto Real Time Clock Card")
+
+
```

Comment:
Please comment closing braces of namespaces.

Indentation has gone wrong.

Stray blank lines at the end of the file.

Reply:
Done, done and done.